### ISSUE TITLE
zfs features auto-detection for ZFSOnLinux

### ISSUE THREAD

**Issue by onlime** [bolthole/zrep#2]
Hi Philip,

What a great script! That's exactly what we were looking for when we planned to move away from our existing rsync-based solution under ext4 on Proxmox VE (Debian Linux based virtualization platform for OpenVZ & KVM). Very nice, you're providing this great script to the community. Thanks.
We are not yet running ZFS/ZREP in production but the flags are all green to start soon, since Proxmox VE 3.4 introduced ZFS support with the latest [ZFSOnLinux](http://zfsonlinux.org/) built in.

One thing that did not work on ZFSOnLinux, obviously was the detection of Z_HAS_X / Z_HAS_SNAPPROPS / DEPTHCAP in [zrep_vars](https://github.com/bolthole/zrep/blob/master/zrep_vars), lines 45-78. As a quickfix, we were simply overriding the (wrongly) auto-detected values:

```ksh
Z_HAS_X=0         # cannot use recv -x
Z_HAS_SNAPPROPS=1 # can set properties on snapshots
DEPTHCAP="-d 1"   # limits "list -r"
```
If you could also make the auto-detection of these features compatible with zfs output of ZFSOnLinux, that would be great. Here you go:

```
$ zfs 2>&1
missing command
usage: zfs command args ...
where 'command' is one of the following:

	create [-p] [-o property=value] ... <filesystem>
	create [-ps] [-b blocksize] [-o property=value] ... -V <size> <volume>
	destroy [-fnpRrv] <filesystem|volume>
	destroy [-dnpRrv] <filesystem|volume>@<snap>[%<snap>][,...]

	snapshot|snap [-r] [-o property=value] ... <filesystem@snapname|volume@snapname> ...
	rollback [-rRf] <snapshot>
	clone [-p] [-o property=value] ... <snapshot> <filesystem|volume>
	promote <clone-filesystem>
	rename [-f] <filesystem|volume|snapshot> <filesystem|volume|snapshot>
	rename [-f] -p <filesystem|volume> <filesystem|volume>
	rename -r <snapshot> <snapshot>
	list [-Hp] [-r|-d max] [-o property[,...]] [-s property]...
	    [-S property]... [-t type[,...]] [filesystem|volume|snapshot] ...

	set <property=value> <filesystem|volume|snapshot> ...
	get [-rHp] [-d max] [-o "all" | field[,...]] [-t type[,...]] [-s source[,...]]
	    <"all" | property[,...]> [filesystem|volume|snapshot] ...
	inherit [-rS] <property> <filesystem|volume|snapshot> ...
	upgrade [-v]
	upgrade [-r] [-V version] <-a | filesystem ...>
	userspace [-Hinp] [-o field[,...]] [-s field]...
	    [-S field]... [-t type[,...]] <filesystem|snapshot>
	groupspace [-Hinp] [-o field[,...]] [-s field]...
	    [-S field]... [-t type[,...]] <filesystem|snapshot>

	mount
	mount [-vO] [-o opts] <-a | filesystem>
	unmount [-f] <-a | filesystem|mountpoint>
	share <-a | filesystem>
	unshare <-a | filesystem|mountpoint>

	send [-DnPpRrv] [-[iI] snapshot] <snapshot>
	receive [-vnFu] <filesystem|volume|snapshot>
	receive [-vnFu] [-d | -e] <filesystem>

	allow <filesystem|volume>
	allow [-ldug] <"everyone"|user|group>[,...] <perm|@setname>[,...]
	    <filesystem|volume>
	allow [-ld] -e <perm|@setname>[,...] <filesystem|volume>
	allow -c <perm|@setname>[,...] <filesystem|volume>
	allow -s @setname <perm|@setname>[,...] <filesystem|volume>

	unallow [-rldug] <"everyone"|user|group>[,...]
	    [<perm|@setname>[,...]] <filesystem|volume>
	unallow [-rld] -e [<perm|@setname>[,...]] <filesystem|volume>
	unallow [-r] -c [<perm|@setname>[,...]] <filesystem|volume>
	unallow [-r] -s @setname [<perm|@setname>[,...]] <filesystem|volume>

	hold [-r] <tag> <snapshot> ...
	holds [-r] <snapshot> ...
	release [-r] <tag> <snapshot> ...
	diff [-FHt] <snapshot> [snapshot|filesystem]

Each dataset is of the form: pool/[dataset/]*dataset[@name]

For the property list, run: zfs set|get

For the delegated permission list, run: zfs allow|unallow
```

Best regards,
Philip

**Comment by ppbrown** [comment]
Howdy,

please let me know first of all that you DO get this :)

after that, I hope you will be my guineapigs :D
grab the latest commit, then do a make, to generate a non-released alpha
version of zrep 1.2.

i think it should autodetect properly for you

**Comment by onlime** [comment]
Wow, you're da man!
I knew you were not born as a Philip for nothing. You are a great honor for our name. :)

Your non-released alpha 1.2 (what a wonderful name...) works great! Tested on Proxmox VE 3.4 (Debian Wheezy) with latest stable ZFSOnLinux (v0.6.3, ZFS pool version 5000, ZFS filesystem version 5). My debugging returned the correctly detected values and everything works smooth:

```
Z_HAS_X: 0
Z_HAS_SNAPPROPS: 1
DEPTHCAP: -d 1
```

I am not terribly happy about this error prone detection by these simple grep commands. But hey, it works and hopefully zfs help output won't change every day.

Thanks! Cheers, Philip
